### PR TITLE
Enable reporting gRPC codes as labels in cortex and ingester_client request duration metrics

### PR DESCRIPTION
**Pull request by duricanikolic**
#### What this PR does

This PR does the following changes:
- Updates vendored `dskit` to revision [b3823cb](https://github.com/grafana/dskit/commit/b3823cb5cdb4239fa5c9fafc57581d584977dbf5). This version of `dskit` introduces `-server.report-grpc-codes-in-instrumentation-label` CLI flag that, when set to true enables reporting gRPC codes as values of the `status_code` label in `cortex_request_duration_seconds` metrics. It defaults to false, meaning that successful and erroneous gRPC status codes are represented with `success` and `error` respectively.
- Introduces `-ingester.client.report-grpc-status-codes` CLI flag that, when set to true enables reporting gRPC codes as values of the `status_code` label in `cortex_ingester_client_request_duration_seconds` metrics. It defaults to false, meaning that successful and erroneous gRPC status codes are represented with `2xx` and `error` respectively.

#### Which issue(s) this PR fixes or relates to

Part of https://github.com/grafana/mimir/issues/6008

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/ingester/client/client.go [hunk 1]
```diff
@@ -64,14 +69,16 @@ func (c *closableHealthAndIngesterClient) Close() error {
 
 // Config is the configuration struct for the ingester client
 type Config struct {
-	GRPCClientConfig grpcclient.Config    `yaml:"grpc_client_config" doc:"description=Configures the gRPC client used to communicate with ingesters from distributors, queriers and rulers."`
-	CircuitBreaker   CircuitBreakerConfig `yaml:"circuit_breaker"`
+	GRPCClientConfig      grpcclient.Config    `yaml:"grpc_client_config" doc:"description=Configures the gRPC client used to communicate with ingesters from distributors, queriers and rulers."`
+	CircuitBreaker        CircuitBreakerConfig `yaml:"circuit_breaker"`
+	ReportGRPCStatusCodes bool                 `yaml:"report_grpc_status_codes"`
```
Comment (pstibrany) [inline pkg/ingester/client/client.go:74]: I'd suggest to mark this as `category:"advanced"`.

**Review by pstibrany — APPROVED** [review]
Thanks, PR looks good to me, but reviewing this made me realize that we named the flag in dskit poorly. It should have been with `-enabled` suffix, so for example: `-server.report-grpc-codes-in-instrumentation-label-enabled`.

Similarly new client flag should be `-ingester.client.report-grpc-status-codes-enabled` (or to be consistent with server: `-ingester.client.report-grpc-codes-in-instrumentation-label-enabled`).

Can we please fix name in dskit?

File: pkg/ingester/client/client.go (hunk 1 above)
Comment (duricanikolic) [reply pkg/ingester/client/client.go]: I have renamed the CLI flag to `-ingester.client.report-grpc-codes-in-instrumentation-label-enabled` and marked it as advanced.

**Review by pstibrany — APPROVED** [review]
Thanks.

**Pull request merged**